components/brave_wallet/common/string_utils.cc

```diff
@@ -10,80 +10,25 @@
 #include <string_view>
 
 #include "base/numerics/safe_math.h"
+#include "base/strings/string_number_conversions_internal.h"
 #include "base/strings/string_util.h"
 
 namespace brave_wallet {
 
-// Determines if the passed in base-10 string is valid
-bool IsValidBase10String(std::string_view input) {
-  if (input.empty()) {
-    return false;
-  }
-  auto check_input =
-      !input.empty() && input.front() == '-' ? input.substr(1) : input;
-  for (const auto& c : check_input) {
-    if (!base::IsAsciiDigit(c)) {
-      return false;
-    }
-  }
-  return true;
-}
-
 std::optional<uint256_t> Base10ValueToUint256(std::string_view input) {
-  if (!IsValidBase10String(input)) {
-    return std::nullopt;
-  }
-
-  base::CheckedNumeric<uint256_t> out = 0;
-  for (char c : input) {
-    out *= 10u;
-    out += static_cast<uint8_t>(base::HexDigitToInt(c));
-  }
-
-  if (!out.IsValid()) {
-    return std::nullopt;
-  }
-  return out.ValueOrDie();
+  uint256_t output;
+  return base::internal::StringToIntImpl(input, output) ? std::optional(output)
```

Comment:
I was recently discouraged to do something with base::internal
https://github.com/brave/brave-core/pull/26253/files/3c3444a73dd13b6bb8d66be4b18005b8b24bafb7#r1831305013
So it is ok or not? @bridiver

Reply:
You were also unnecessarily inheriting from `std::array`, which was unnecessarily breaking the extent computation, just to clarify why specialising that extent was not supposed to have happened.

Reply:
👍